### PR TITLE
Adds tests for PR 5659

### PR DESCRIPTION
**Pull request by sklam**
Based on #5659 
Adds tests for 1D, 2D, 3D and C/F order arrays for parfors.

**Note for reviewer**: the main changes to the PR starts in https://github.com/numba/numba/pull/6664/commits/2628ec4b541b77fe4ce6dfe19079f9335c836704.
The only file I touched in the `test_parfors.py`.

### CODE REVIEW

File: numba/tests/test_parfors.py [hunk 1]
```diff
@@ -529,16 +530,60 @@ def check(self, pyfunc, *args, **kwargs):
...
+        """
+        def gen():
+            yield np.linspace(0, 1, n)
+            yield np.linspace(2, 1, n)
+            yield np.linspace(1, 2, n)
+
+        src = cycle(gen())
+        return [next(src) for i in range(ct)]
+
+    def gen_linespace_variants(self, ct):
```
Comment (DrTodd13) [inline numba/tests/test_parfors.py:544]: Did you mean linespace or linspace?

**Review by DrTodd13 — CHANGES_REQUESTED** [review]
This needed to be done.  Thanks for doing it.  Looks like it is running with no errors.

File: numba/tests/test_parfors.py (hunk 1 above)
Comment (sklam) [reply numba/tests/test_parfors.py:544]: oops, it's linspace.

File: numba/parfors/parfor.py [hunk 2]
```diff
@@ -1920,7 +1920,8 @@ def run(self, blocks):
                 if isinstance(instr, ir.Assign):
                     expr = instr.value
                     lhs = instr.target
-                    if self._is_C_order(lhs.name):
+                    lhs_typ = self.pass_states.typemap[lhs.name]
+                    if self._is_C_or_F_order(lhs_typ):
                         # only translate C order since we can't allocate F
```
Comment (stuartarchibald) [inline numba/parfors/parfor.py:1925]: remove this comment?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for writing these extra tests @sklam, looks good, just one dead comment to remove and then can be merged.

File: numba/parfors/parfor.py (hunk 2 above)
Comment (DrTodd13) [reply numba/parfors/parfor.py:1925]: Yes, this comment should go.

**Review by DrTodd13 — DISMISSED** [review]

**Comment by sklam** [comment]
i've removed the dead comment

**Comment by stuartarchibald** [comment]
> i've removed the dead comment

thanks.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch(es). Looks good, great to see this working.

**Pull request merged**